File: string/reverse.js
// Program to reverse each word in a sentence

function reverseBySeparator(str, separator) {
	return str.split(separator).reverse().join(separator);
}

// test case
var testString = "Hey dude!";
// reverse entire sentence
var reverseSentence = reverseBySeparator(testString, ""); // expect to return !edud yeH

// reverse each word
reverseBySeparator(reverseSentence, " "); // expect to return yeH !edud

File: string/reverse.py
# Reverse each word in a sentence

def reverse_sentence(string, separator):
	# string_list = string.split()
	# flipped_list = string_list[::-1]
	flipped_list = (string.split())[::-1] # split string into list and then reverse order of elements in list
	output = separator.join(flipped_list)

	print output
